aesara/tensor/random/type.py

```diff
@@ -14,26 +18,52 @@ class RandomStateType(Type):
     with the `==` operator.  This `Type` exists to provide an equals function
     that is used by `DebugMode`.
 
+    Also works with a `dict` derived from RandomState.get_state() unless
+    the `strict` argument is explicitly set to `True`.
+
     """
 
     def __repr__(self):
         return "RandomStateType"
 
     @classmethod
     def filter(cls, data, strict=False, allow_downcast=None):
-        if cls.is_valid_value(data):
+        if cls.is_valid_value(data, strict):
             return data
         else:
             raise TypeError()
 
     @staticmethod
-    def is_valid_value(a):
-        return isinstance(a, np.random.RandomState)
+    def is_valid_value(a, strict):
+        if isinstance(a, np.random.RandomState):
+            return True
+
+        if not strict and isinstance(a, dict):
+            gen_keys = [("gauss", float), ("has_gauss", int)]
+            state_keys = [("key", np.uint32), ("pos", int)]
+
+            for key, key_dtype in gen_keys:
+                if key in a and np.array(a[key]).dtype == key_dtype:
```

Comment:
I think the issue with the failing tests has something to do with the `dtype` checks. I also tried the `np.int/float` dtypes but it isn't working. I cannot replicate this issue locally so kind of hard to debug this from here.